### PR TITLE
Adding metadata getter methods to datasets API

### LINKED ISSUE #3820: [FEA]: Add getters to Datasets API to easily get metadata information
### Is this a new feature, an improvement, or a change to existing functionality?

Improvement

### How would you describe the priority of this feature request

Low (would be nice)

### Please provide a clear description of problem this feature solves

If a cugraph user wants to view metadata (eg number of nodes) in a dataset, they have to either navigate to the metadata path to view the YAML files or access it through `dataset.metadata`.

### Describe your ideal solution

To simplify the process for the end-user, Datasets objects can have getter methods that quickly read from the metadata.

```python3
# old
dataset.metadata['is_directed']
# new
dataset.is_directed()
```

### Describe any alternatives you have considered

n/a

### Additional context

n/a

### Code of Conduct

- [X] I agree to follow cuGraph's Code of Conduct
- [X] I have searched the [open feature requests](https://github.com/rapidsai/cugraph/issues?q=is%3Aopen+is%3Aissue+label%3A%22feature+request%22%2Cimprovement%2Cenhancement) and have found no duplicates for this feature request

### PR DESCRIPTION
**Pull request by nv-rliu**
Closes #3820 

This PR adds simple getter methods to the `dataset` class, which allows users to easily get information about datasets without need to access the `metadata` dict or look in the directory.

```python
from cugraph.datasets import karate

# users now call
karate.number_of_nodes()

# instead of 
karate.metadata['number_of_nodes']
```

### CODE REVIEW

**Review by alexbarghi-nv — APPROVED** [review]
👍

File: python/cugraph/cugraph/tests/utils/test_dataset.py [hunk 1]
```diff
@@ -328,6 +328,16 @@ def test_is_multigraph(dataset):
...
 
 
+@pytest.mark.parametrize("dataset", ALL_DATASETS)
+def test_object_getters(dataset):
+    assert dataset.is_directed() == dataset.metadata["is_directed"]
+    assert dataset.is_multigraph() == dataset.metadata["is_multigraph"]
+    assert dataset.is_symmetric() == dataset.metadata["is_symmetric"]
+    assert dataset.number_of_nodes() == dataset.metadata["number_of_nodes"]
+    assert dataset.number_of_vertices() == dataset.metadata["number_of_nodes"]
+    assert dataset.number_of_edges() == dataset.metadata["number_of_edges"]
```
Comment (rlratzel) [inline python/cugraph/cugraph/tests/utils/test_dataset.py:338]: This really isn't testing much more than if the APIs are returning the metadata dict, and could start failing if we ever intentionally change the implementation to return something that's still valid but doesn't match the metadata dict exactly.

A better test would be if you verified `is_directed` and others are actually True or False based on if the datasets are directed or not, etc.

**Pull request merged**